src/app/pages/forms-and-validation/phone-number-format/phone-number-format.component.ts

```diff
@@ -0,0 +1,120 @@
+        { code: 'RU', name: '+7 (RU)', placeholder: '### ### ## ##', maxLength: '13', errorCode: 'Russian' },
+        { code: 'EG', name: '+20 (EG)', placeholder: '# #######', maxLength: '9', errorCode: 'Egyptian' },
+        { code: 'IN', name: '+91 (IN)', placeholder: '#### ### ###', maxLength: '12', errorCode: 'Indian' },
+    ];
+    selectedCountry = this.countries[0];
+    countryChange = false;
+
+    constructor(
+        private readonly _drawerService: StateService,
+        private readonly _breakpointObserver: BreakpointObserver,
+        private readonly _formBuilder: FormBuilder
+    ) {
+        this.initForm();
+    }
+
+    ngOnInit(): void {
+        this._breakpointObserver
+            .observe([Breakpoints.Small, Breakpoints.Handset])
+            .subscribe((state: BreakpointState) => {
+                if (state.matches) {
+                    this.isSmall = true;
+                } else {
+                    this.isSmall = false;
+                }
+            });
+    }
+
+    initForm(): void {
+        this.validatePhoneNumberForm = this._formBuilder.group({
+            phone: ['', [Validators.required, this.checkPhoneNumber.bind(this)]],
+            selectedCountry: ['US', Validators.required],
+        });
+    }
+
+    checkPhoneNumberPattern(phoneNumber: string, countryCode: string): boolean {
+        switch (countryCode) {
+            case 'US':
+            case 'CA': {
+                return /^\(?([0-9]{3})\)?[-. ]?([0-9]{3})[-. ]?([0-9]{4})$/.test(phoneNumber);
+            }
+            case 'RU': {
+                return /^\(?([0-9]{3})\)?[-. ]?([0-9]{3})[-. ]?([0-9]{2})[-. ]?([0-9]{2})$/.test(phoneNumber);
+            }
+            case 'EG': {
+                return /^\(?([0-9]{1})\)?[-. ]?([0-9]{7})$/.test(phoneNumber);
+            }
+            case 'IN': {
+                return /^\(?([0-9]{4})\)?[-. ]?([0-9]{3})[-. ]?([0-9]{3})$/.test(phoneNumber);
+            }
+            default: {
+                return /^\(?([0-9]{3})\)?[-. ]?([0-9]{3})[-. ]?([0-9]{4})$/.test(phoneNumber);
+            }
+        }
+    }
+
+    checkPhoneNumber(phoneNumberInput: AbstractControl): any {
+        const isValid = this.checkPhoneNumberPattern(phoneNumberInput.value, this.selectedCountry.code);
+        return isValid ? null : { phoneNumberNotMatch: true };
+    }
+
+    validatePhoneNumber(): void {
+        if (this.validatePhoneNumberForm.invalid) {
+            this.validatePhoneNumberForm.controls['phone'].setErrors({ phoneNumberNotMatch: true });
+        }
+    }
+
+    removeError(): void {
+        this.validatePhoneNumberForm.controls.phone.markAsUntouched();
+    }
+
+    allowNumbersOnly(event: KeyboardEvent): void {
+        const allowedRegex = /[0-9]/g;
+        if (!event.key.match(allowedRegex)) {
+            event.preventDefault();
+        }
+        this.countryChange = false;
+    }
```

Comment:
Is there a reason you are using this method instead of using the `matInput type="number"`?
https://material.angular.io/components/input/overview#supported-input-types

Reply:
I tried with input type="number", it shows arrows which are not matching our design so decided to use type="text". 
![image](https://user-images.githubusercontent.com/10433274/130184735-f0847647-32ad-4e61-bb16-511bfb9a6492.png)

If we want type-"number" then I will try to hide arrow using CSS.

Reply:
@huayunh Any thoughts on this `type="number"` arrow dilemma? I don't mind leaving it as a text input with the custom function, just feels like overkill when there is a number type option.